_minutes/2023-09-11-2023-09-14-tpac-extra.md

```diff
@@ -0,0 +1,520 @@
+
+
+### Topics for discussion
+
+ * [oliver] Alarms: What to do when the laptop closes
+   * [WECG Issue 433](https://github.com/w3c/webextensions/issues/433)
+   * Two issues
+     * Can create an alarm in two ways: delay in minutes or "when"
+       * If you put a delay of 5 minutes, then, after two minutes, put the device to sleep for two minutes, you need to wait 3 more minutes for it to fire
+       * Happens for both "when" and "delayInMinutes"
+     * Multiple extension alarms firing can cause recalculation, not taking into account device sleep
+       * Non-controversial: Extension A's alarms should never affect Extension B
+     * [rob] periodInMinutes minimum is 1. Is one-minute resolution a hard requirement, or is a lower resolution acceptable?
+       * [devlin] Let's table for now.
+     * [devlin]: Also non-controversial, firing for "when" should always fire at the time, not take into account device sleep
+     * [devlin]: What about for delayInMinutes?  Around the table.
+       * Unanimous consensus: should be wall clock
+     * [devlin] Edge cases!
+       * [rob] Device is asleep at the time
+         * [tom] Delay in minutes has passed; should obviously run
+         * [devlin] Should they?  Three possibilities
+           * Fire them
+           * Don't fire them
+           * Fire them with a different (expired alarm) event
+         * [oliver] With delayInMinutes, should fire, with "when" should probably still fire
+         * [devlin] Devil's advocate: if it's time-sensitive, you may want to change handling.  onExpiredAlarm would be more clear
+         * [oliver] It's easy enough to just calculate it
+         * [rob] could introduce a new option
+         * [devlin] eww.  Also not _that_ worried about breaking changes for this scenario, since it is already very broken for device asleep cases
+         * [kiara] aligned on should fire on device wakeup?
+         * [simeon] What's simplest to grok?  Firing immediately
+         * [rob] What about periodInMinutes? (repeating timer)
+         * [oliver] Ugh.  We shouldn't fire 100 alarms for 100 missed periods
+         * [simeon] I would expect periodInMinutes to fire at least &lt;N> minutes after last firing period.
+         * [devlin] But then you get significant drift.  That was a past bug in Chromium we fixed.
+         * [oliver] align with setInterval behavior if possible?  What is setInterval behavior?
+         * [devlin] Or setTimeout behavior while laptop closed?
+         * [oliver] Thinking about a use case with a 24 hour timer (e.g. syncing), we wouldn't want that to not fire just because you missed it by a minute
+         * [devlin] Web Platform Behavior (in FF and Chrome):
+           setInterval fires once for missed interval
+           setTimeout fires for missed timeout
+         * [tom] Unanimous that we should align with setTimeout and setInterval should behave similar ot the web?
+         * We think so, but some spec questions &lt;Oliver to add>
```

Comment:
@oliverdunk Do you recall what you wanted to add here?

Reply:
```suggestion
         * [oliver] Looking at the spec unfortunately seems to contradict what Devlin saw in testing. We should speak to some people familiar with the spec to decide next steps.
```